### PR TITLE
MAINT - Improvements

### PR DESCRIPTION
**Pull request by blen-desta**
### Description of the Change
- Added a help option and extracted the test running into its own class.
- Added javadoc to the RestAssuredExtensions class and a check that the response body is not null if the binding is not redirect.

See: https://github.com/codice/saml-conformance/issues/209

screen shot 2018-12-17 at 10 25 21 am

### SAML V2.0 Specification Quote
N/A

### Verification Process
`./samlconf --help` displays the script options 

### Checklist:
- [ ] Unit Tests Added/Modified

### CODE REVIEW

File: deployment/distribution/src/main/kotlin/org/codice/ctk/Command.kt [hunk 1]
```diff
@@ -14,141 +14,84 @@ import org.codice.compliance.LENIENT_ERROR_VERIFICATION
...
+    val defaultImplPath = "$samlDist${File.separator}$DEFAULT_IMPLEMENTATION_PATH"
+    val ctkMetadataPath = "$samlDist${File.separator}conf${File.separator}samlconf-sp-metadata.xml"
+
     val parser = createParser()
     val arguments = parser.parse(args)
 
-    val implementationPath = arguments.option("i")
-            ?: "$samlDist${File.separator}$DEFAULT_IMPLEMENTATION_PATH"
+    val implementationPath = arguments.option("i") ?: defaultImplPath
     val userLogin = arguments.option("i") ?: "admin:admin"
```
Comment (Kjames5269) [inline deployment/distribution/src/main/kotlin/org/codice/ctk/Command.kt]: ❗️Wasn't modified by this PR but this should be "u"

File: deployment/distribution/src/main/kotlin/org/codice/ctk/Command.kt [hunk 2]
```diff
@@ -14,141 +14,84 @@ import org.codice.compliance.LENIENT_ERROR_VERIFICATION
...
 
+    if (arguments.flag("help")) {
+        println(parser.printHelp())
```
[the remark this answers is not in the document]
Comment (blen-desta) [reply deployment/distribution/src/main/kotlin/org/codice/ctk/Command.kt]: It doesn't do so much. Plus the next version breaks everything ¯\_(ツ)_/¯

File: deployment/distribution/src/main/kotlin/org/codice/ctk/Command.kt [hunk 3]
```diff
@@ -14,141 +14,84 @@ import org.codice.compliance.LENIENT_ERROR_VERIFICATION
...
+        return
+    }
+
+    println("u=${arguments.option("u")}")
+    println("userLogin=${arguments.option("userLogin")}")
```
[the remark this answers is not in the document]
Comment (blen-desta) [reply deployment/distribution/src/main/kotlin/org/codice/ctk/Command.kt]: Oops that was when I was testing things

File: deployment/distribution/src/main/kotlin/org/codice/ctk/Command.kt [hunk 4]
```diff
@@ -14,141 +14,84 @@ import org.codice.compliance.LENIENT_ERROR_VERIFICATION
...
     System.setProperty(LENIENT_ERROR_VERIFICATION, arguments.flag("l").toString())
     System.setProperty(RUN_DDF_PROFILE, arguments.flag("ddf").toString())
 
-    if (arguments.flag("debug")) {
-        Log.logLevel = LogLevel.DEBUG
-    } else {
-        Log.logLevel = LogLevel.INFO
-    }
+    if (arguments.flag("debug")) Log.logLevel = LogLevel.DEBUG
+    else Log.logLevel = LogLevel.INFO
```
[the remark this answers is not in the document]
Comment (blen-desta) [reply deployment/distribution/src/main/kotlin/org/codice/ctk/Command.kt]: ¯\_(ツ)_/¯ I'll change that

File: external/implementations/samlconf-ddf-impl/src/main/resources/ddf-idp-metadata.xml [hunk 5]
```diff
@@ -1,4 +1,4 @@
-<md:EntityDescriptor xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata" cacheDuration="P7DT0H0M0.000S" entityID="https://localhost:8993/services/idp/login">
+<md:EntityDescriptor xmlns:md="urn:oasis:names:tc:SAML:2.0:metadata" cacheDuration="P7DT0H0M0.000S" entityID="https://localhost:32781/ddf/services/idp/login">
```
[the remark this answers is not in the document]
Comment (blen-desta) [reply external/implementations/samlconf-ddf-impl/src/main/resources/ddf-idp-metadata.xml]: No idea. Didn't even know I changed that

File: deployment/distribution/src/main/kotlin/org/codice/ctk/Command.kt (hunk 1 above)
Comment (blen-desta) [reply deployment/distribution/src/main/kotlin/org/codice/ctk/Command.kt]: 👌

**Comment by blen-desta** [comment]
build now

**Comment by blen-desta** [comment]
build now

**Comment by blen-desta** [comment]
test ddf

**Pull request merged**